tools/sof_ri_info/sof_ri_info.py

```diff
@@ -0,0 +1,883 @@
+        self.data = open(path, 'rb').read()
+        self.file_size = len(self.data)
+        if self.verbose:
+            print('File size ' + uint_to_string(self.file_size, True))
+        self.cur_offset = 0
+
+    def get_offset(self):
+        """ Retrieve the offset, the reader is at
+        """
+        return self.cur_offset
+
+    def ff_data(self, delta_offset):
+        """ Forwards the read pointer by specified number of bytes
+        """
+        self.cur_offset += delta_offset
+
+    def set_offset(self, offset):
+        """ Set current reader offset
+        """
+        old_offset = self.cur_offset
+        self.cur_offset = offset
+        return old_offset
+
+    def get_data(self, beg, length):
+        """ Retrieves the data from beg to beg+length.
+            This one is good to peek the data w/o advancing the read pointer
+        """
+        return self.data[self.cur_offset +beg : self.cur_offset +beg +length]
+
+    def read_bytes(self, count):
+        """ Reads the specified number of bytes from the stream
+        """
+        bts = self.get_data(0, count)
+        self.ff_data(count)
+        return bts
+
+    def read_dw(self):
+        """ Reads a dword from the stream
+        """
+        dword, = struct.unpack('I', self.get_data(0, 4))
+        self.ff_data(4)
+        return dword
+
+    def read_w(self):
+        """ Reads a word from the stream
+        """
+        word, = struct.unpack('H', self.get_data(0, 2))
+        self.ff_data(2)
+        return word
+
+    def read_b(self):
+        """ Reads a byte from the stream
+        """
+        byte, = struct.unpack('B', self.get_data(0, 1))
+        self.ff_data(1)
+        return byte
+
+    def read_uuid(self):
+        """ Reads a UUID from the stream and returns as string
+        """
+        out = '{:08x}'.format(self.read_dw())
+        out += '-'+'{:04x}'.format(self.read_w())
+        out += '-'+'{:04x}'.format(self.read_w())
+        out += '-'+'{:02x}'.format(self.read_b()) + \
+               '{:02x}'.format(self.read_b()) + '-'
+        for _ in range(0, 6):
+            out += '{:02x}'.format(self.read_b())
+        return out
+
+    def offset_to_string(self, delta=0):
+        """ Retrieves readable representation of the current offset value
+        """
+        return uint_to_string(self.cur_offset+delta)
+
+    def info(self, loginfo, off_delta=0, verb_info=True):
+        """ Prints 'info' log to the output, respects verbose mode
+        """
+        if verb_info and not self.verbose:
+            return
+        print(self.offset_to_string(off_delta) + '\t' + loginfo)
+
+    def error(self, logerror, off_delta=0):
+        """ Prints 'error' log to the output
+        """
+        print(self.offset_to_string(off_delta) + '\terror: ' + logerror)
+
+# Data Model
+
+class Attribute():
+    """ Attribute: base class with global formatting options
+    """
+    no_colors = False
+    full_bytes = True
+
+class Auint(Attribute):
```

Comment:
IMHO the Auint etc classes should be names UIntAttribute, HexAttribute etc, A* names are a bit obscure

Reply:
I hope both the definition which includes base name and use context are obvious enough to keep the names compacted in this particular case.